packages/reactivity/src/baseHandlers.ts

```diff
@@ -86,7 +86,12 @@ class BaseReactiveHandler implements ProxyHandler<Target> {
 
     if (!isReadonly) {
       let fn: Function | undefined
-      if (targetIsArray && (fn = arrayInstrumentations[key])) {
+      if (
+        targetIsArray &&
+        // @ts-expect-error our code is limited to es2016 but user code is not
+        (target as any[])[key] === Array.prototype[key] &&
```

Comment:
This will mean that extended methods lose tracking optimization

Reply:
I'm sorry, but I don't know what "tracking optimization" means. Is it referring to this part?
![image](https://github.com/user-attachments/assets/c209f177-9f12-4d85-aeda-8ee6845f753e)

Reply:
see #9511

Reply:
If that's the case, the extended methods indeed lose tracking optimization.

However, I'm puzzled by the mismatch between the variable values and the function's parameter definition during debugging. Although the problem was solved by using `arguments`.

![image](https://github.com/user-attachments/assets/f5e0290a-b91a-46a7-ac4d-5cf75d94ccef)